### DISCUSSION TITLE
When can trino-hive-apache support java17 ?

### DISCUSSION THREAD

**Discussion by jiangjiguang** [trinodb/trino#15310] [Q&A]
Since trino has supported java17, but now trino-hive-apache(https://github.com/trinodb/trino-hive-apache) errors when compiled on java17.

**Comment by dain** [comment]
Why would you want to compile trino-hive-apache with Java 17? The produced library will run on 17, so I don't seen much advantage switching this right now.

**Comment by jiangjiguang** [comment]
> Why would you want to compile trino-hive-apache with Java 17? The produced library will run on 17, so I don't seen much advantage switching this right now.

Because I have done optim on apache parquet with java17 vector api. trino-hive-apache has used apache parquet.
